packages/core/js-sdk/src/auth/index.ts

```diff
@@ -0,0 +1,33 @@
+import { Client } from "../client"
+import { Config } from "../types"
+
+export class Auth {
+  private client: Client
+  private config: Config
+
+  constructor(client: Client, config: Config) {
+    this.client = client
+    this.config = config
+  }
+
+  login = async (payload: { email: string; password: string }) => {
+    // TODO: It is a bit strange to eg. require to pass `scope` in `login`, it might be better for us to have auth methods in both `admin` and `store` classes instead?
```

Comment:
I believe it would be easier to use as well as we would be inferring the scope ourselves, and if the user want a global auth with scope we can either provide it (in case the user customisations involves more scopes) or they can extends the client themselves no?

Reply:
@adrien2p yeah that's what I am thinking, but let's see what others think as well.

Reply:
I don't mind having auth in `store.` and `admin.` - but would still want there to be an `auth.` so you can access that endpoint easily if you have custom scopes.

Reply:
@srindom yeah makes sense, we can do that. `auth` can expose the more generic endpoint, whereas store and admin will simply call it with hard-coded scopes. I'll do it in a follow-up PR